tests/unit/recommenders/models/test_sar_singlenode.py

```diff
@@ -413,3 +418,43 @@ def test_match_similarity_type_from_json_file(header):
 
     # make sure fit still works when similarity type is loaded from a json file
     model.fit(train)
+
```

Comment:
Can we add a test that fails when there are duplicates?
```
def test_dataset_with_duplicates(header):

    model = SARSingleNode(**header)
    train = pd.DataFrame(
        {
            header["col_user"]: [1, 1, 2, 2, 2],
            header["col_item"]: [1, 2, 1, 2, 2],
            header["col_rating"]: [3.0, 4.0, 3.0, 4.0, 4.0]
        }
    )
with pytest.raises(ValueError):
        model.fit(train)
    
```

Reply:
good call. added.